Release v0.8

tym2k1 opened the pull request:
Merged PRs:
https://github.com/3mdeb/meta-rte/pull/69
https://github.com/3mdeb/meta-rte/pull/70
https://github.com/3mdeb/meta-rte/pull/71 counterpart with adjustments - see https://github.com/3mdeb/meta-rte/pull/73/commits/200cf5ba178d2f65ba795a2d3e32f545abbecda4
https://github.com/3mdeb/meta-rte/pull/76

tym2k1 committed:
*: updated bitbake syntax

See: https://git.yoctoproject.org/poky/tree/scripts/contrib/convert-overrides.py

tym2k1 committed:
conf: changed layer compatibility to scarthgap

tym2k1 committed:
kas.yml: updated fetched layers revisions to scarthgap

tym2k1 committed:
kas.yml: added meta-arm dependancy

See: https://github.com/linux-sunxi/meta-sunxi/blob/scarthgap/conf/layer.conf#L15

tym2k1 committed:
kas.yml: removed image-mklibs & image-prelink

Related patch notes:
- https://docs.yoctoproject.org/migration-guides/migration-3.4.html?highlight=mklibs#removed-classes
- https://docs.yoctoproject.org/migration-guides/migration-4.0.html#prelink-removed

tym2k1 committed:
patches/0001-do-not-remove-wic-partitions.patch: updated

tym2k1 committed:
packagegroups/packagegroup-rte.bb: bmap-tools -> bmaptool

See: https://git.yoctoproject.org/poky/commit/?id=f15676592bb8ba082b31d35c46631f35f1b3a222

tym2k1 committed:
conf/distro/rte.conf: added usrmerge to DISTRO_FEATURES

See: https://docs.yoctoproject.org/migration-guides/migration-4.3.html?highlight=usrmerge#systemd-changes

tym2k1 committed:
conf/distro/rte.conf: bump used u-boot version

tym2k1 committed:
u-boot/0004-sun8i-disable-Vendor-Parameter-Protection.patch: updated

tym2k1 committed:
recipes-devtools/dediprog-flasher: add recipe

tym2k1 committed:
recipes-extended/packagegroups/packagegroup-rte.bb: added dediprog-flasher

tym2k1 committed:
dediprog-flasher/0001-add-support-for-cross-compilation.patch: added

tym2k1 committed:
recipes-devtools/dediprog-flasher: do_install() add PREFIX=/usr

tym2k1 committed:
dediprog-flasher/0002-Makefile-remove-stripping.patch: added

tym2k1 committed:
recipes-devtools/dediprog-flasher: add /usr/share/DediProg to FILES

tym2k1 committed:
recipes-bsp/u-boot/u-boot/*: added Upstream-Status to patches

tym2k1 committed:
u-boot/0001-nanopi_neo_air_defconfig-Enable-eMMC-support.patch: removed

Already in [meta-sunxi](https://github.com/linux-sunxi/meta-sunxi/blob/master/recipes-bsp/u-boot/files/0001-nanopi_neo_air_defconfig-Enable-eMMC-support.patch)

tym2k1 committed:
recipes-bsp/u-boot/u-boot/*: updated context lines to fix fuzz

```
devtool modify u-boot
devtool finish --force-patch-refresh u-boot /build/../repo
```

tym2k1 committed:
recipes-bsp/u-boot/u-boot/*: reordered patches

tym2k1 committed:
kas.yml: ABORT -> HALT

tym2k1 committed:
recipes-graphics/xorg-proto/xcb-proto: use package from upstream

tym2k1 committed:
github.com-gorilla-mux/github.com-gorilla-mux_git.bb: updated rev

tym2k1 committed:
recipes-go/github.com-*: inherit go -> inherit go-mod

tym2k1 committed:
recipes-go/3mdeb-rtectrl/3mdeb-rtectrl_git.bb: Updated commit

tym2k1 committed:
recipes-devtools/stlink/stlink_git.bb: LICENSE = BSD -> BSD-3-Clause

tym2k1 committed:
recipes-support/openvpn: deleted

Related to https://github.com/3mdeb/meta-rte/issues/72
Also deleted the `/data` partition that was used for
storing config

tym2k1 committed:
images/core-image-minimal.bbappend: updated extrausers syntax

tym2k1 committed:
recipes-devtools/sd-mux/sd-mux_git.bb: changed source

The Tizen project has reached EOL and to ensure that the recipe builds
we forked the source code.

tym2k1 committed:
imx-usb-loader/imx-usb-loader_git.bb: updated LICENSE value

tym2k1 committed:
recipes-*: added missing branch parameters to SRC_URI

tym2k1 committed:
patches/0001-do-not-remove-wic-partitions.patch: mv -> cp

the core-image-minimal's `do_image_task` fails if no image is present as
it can't use `gzip` on it. While everything works fine the script
returns exit code 1 which stops the build.

tym2k1 committed:
recipes-kernel/linux/linux-mainline: updated patches

tym2k1 committed:
minicom: add a default config to disable hardware flow control

Co-authored-by: Wiktor Grzywacz <wiktor.grzywacz@3mdeb.com>

tym2k1 committed:
recipes-devtools/dediprog-flasher: Updated based on meta-oe patch review

Upstream patch: https://patchwork.yoctoproject.org/project/oe/patch/20240722140209.1431735-2-tymoteusz.burak@3mdeb.com/

tym2k1 committed:
distro/rte.conf: bump to 0.8

m-iwanicki commented on README.md:
````diff
@@ -28,14 +27,14 @@ inside container, via the [kas-container] script.
 
-```
+```sh
 wget -O ~/bin/kas-container https://raw.githubusercontent.com/siemens/kas/master/kas-container
````
You should probably use permalink to the version you want e.g.
https://github.com/siemens/kas/blob/422e6032f6a86aa7a3d47671e31f08650e5bd228/kas-container
or
https://raw.githubusercontent.com/siemens/kas/4.4/kas-container
if you want it to be more readable, but tags can be changed so safer would be to use commit hash

tym2k1 replied:
[Adressed](https://github.com/3mdeb/meta-rte/pull/73/commits/a21267c9321973e42d181b8fe1e8ab375e60a782) by parsing env variable with specific version while still using `kas-container` from master branch

m-iwanicki replied:
Have you checked that it works? If yes then :+1:

tym2k1 committed:
README.md: specify which kas version to use

m-iwanicki reviewed (changes requested):
A lot of pre-commit errors need to be fixed.

tym2k1 committed:
*:  Add basic pre-commit configuration

Signed-off-by: Tymoteusz Burak <tymoteusz.burak@3mdeb.com>

tym2k1 committed:
*: Fix files so that pre-commit passes

Signed-off-by: Tymoteusz Burak <tymoteusz.burak@3mdeb.com>

tym2k1 committed:
*: Address pre-commit for non standard recipes

Signed-off-by: Tymoteusz Burak <tymoteusz.burak@3mdeb.com>

tym2k1 commented:
@m-iwanicki pre-commit passes after merging changes from https://github.com/3mdeb/meta-rte/pull/76

tym2k1 commented:
I've removed commits from https://github.com/3mdeb/meta-rte/pull/74 as I copied the old version of pre-commit that used the Zarhus fork and redid all files to adress to new pre-commit config. When building there are some errors with the changed files that I need to adress. Mostly with [non-standard recipes](https://github.com/3mdeb/meta-rte/pull/73/commits/943be9f0f1c8581c51f21b155757e9d9bb2f178f)

m-iwanicki commented:
@tym2k1

> When building there are some errors with the changed files that I need to adress

Are you going to address those errors? PR is marked as ready for review but clearly it isn't.

```
WARNING: Host distribution "debian-12" has not been validated with this version of the build system; you may possibly experience unexpected failures. It is recommended that you use a tested distribution.
Loading cache: 100% |                                                                                                                                          | ETA:  --:--:--
Loaded 0 entries from dependency cache.
WARNING: /build/../work/meta-sunxi/recipes-kernel/xradio-firmware/xradio-firmware.bb: URL: git://github.com/armbian/firmware.git;protocol=https does not set any branch parameter. The future default branch used by tools and repositories is uncertain and we will therefore soon require this is set in all git urls.
Parsing recipes: 100% |#########################################################################################################################################| Time: 0:00:07
Parsing of 2804 .bb files complete (0 cached, 2804 parsed). 4719 targets, 530 skipped, 0 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies
ERROR: Nothing RPROVIDES '"#' (but /build/../work/poky/meta/recipes-core/images/core-image-minimal.bb RDEPENDS on or otherwise requires it)
NOTE: Runtime target '"#' is unbuildable, removing...
Missing or unbuildable dependency chain was: ['"#']
ERROR: Required build target 'core-image-minimal-swu' has no buildable providers.
Missing or unbuildable dependency chain was: ['core-image-minimal-swu', 'core-image-minimal', '"#']

Summary: There were 2 WARNING messages.
Summary: There were 2 ERROR messages, returning a non-zero exit code.
2024-10-23 09:13:16 - ERROR    - Command "/work/poky/bitbake/bin/bitbake -c build core-image-minimal core-image-minimal-swu" failed with error 1
```

PLangowski committed:
core-image-minimal.bbappend: Fix comment

Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>

PLangowski commented:
@m-iwanicki fixed in 2067a858513ea3f2d0ac86c76ff5a3f89db18a92

PLangowski committed:
Fix build errors

Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>

PLangowski commented:
I pushed some fixes. Now the build should succeed.

PLangowski commented:
@tym2k1 The build succeeds, but the system cannot be booted

```
switch to partitions #0, OK
mmc0 is current device
Scanning mmc 0:1...
Found U-Boot script /boot.scr
674 bytes read in 1 ms (658.2 KiB/s)
## Executing script at 43100000
Saving Environment to FAT... OK
24497 bytes read in 3 ms (7.8 MiB/s)
Failed to load '/boot/zImage'
zimage: Bad magic!
Wrong Image Format for bootm command
ERROR: can't get kernel image!
SCRIPT FAILED: continuing...
24497 bytes read in 3 ms (7.8 MiB/s)
BootOrder not defined
EFI boot manager: Cannot load any image

Device 0: unknown device
sun8i_emac_eth_start: Timeout
missing environment variable: pxeuuid
Retrieving file: pxelinux.cfg/01-02-42-50-31-cd-c1
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/00000000
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/0000000
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/000000
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/00000
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/0000
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/000
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/00
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/0
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/default-arm-sunxi-sunxi
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/default-arm-sunxi
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/default-arm
sun8i_emac_eth_start: Timeout
Retrieving file: pxelinux.cfg/default
sun8i_emac_eth_start: Timeout
Config file not found
sun8i_emac_eth_start: Timeout
sun8i_emac_eth_start: Timeout
```

m-iwanicki commented on recipes-core/images/core-image-minimal.bbappend:
```diff
@@ -1,16 +1,11 @@
+    packagegroup-rte-coreboot \
+    locale-base-en-us \
+    "
+# set root passwordINHERIT += "extrausers"RTE_ROOT_PWD = "meta-rte"EXTRA_USERS_PARAMS = "usermod -P ${RTE_ROOT_PWD} root;"
```
@tym2k1 This probably shouldn't be commented out.

macpijan replied:
If that helps, in currently running RTEs the default pwd is `meta-rte` and it would be nice if that persists.

macpijan replied:
@PLangowski @tym2k1 This is probably the last one to be resolved here

macpijan commented on recipes-devtools/dediprog-flasher/dediprog-flasher/0001-add-support-for-cross-compilation.patch:
```diff
@@ -0,0 +1,21 @@
+ 
+ PROGRAM = dpcmd
+-CC      = gcc
++CC     ?= gcc
```
Already merged upstream: https://github.com/DediProgSW/SF100Linux/pull/86/files

macpijan commented on recipes-devtools/dediprog-flasher/dediprog-flasher/0002-Makefile-add-conditional-stripping.patch:
```diff
@@ -0,0 +1,39 @@
+From: Tymoteusz Burak <tymoteusz.burak@3mdeb.com>
+Date: Mon, 22 Jul 2024 13:02:59 +0000
+Subject: [PATCH] Makefile: add conditional stripping
+Upstream-Status: Submitted [https://github.com/DediProgSW/SF100Linux/pull/88]
```
already merged

macpijan replied:
We can remove these patches later, when new release of this tool comes

macpijan commented on recipes-kernel/linux/linux-mainline/0001-sun8i-h2-plus-orangepi-zero.dts-enable-ina219.patch:
```diff
@@ -1,35 +0,0 @@
-
-+&i2c1 {
-+      ina219@40 {
-+              status = "okay";
```
This patch is dropped, do we enable this device somewhere else?

macpijan replied:
It is here: https://github.com/3mdeb/meta-rte/pull/73/files#diff-5b15643f99f3569abf46c083d1734f58498233cac6dae4e2ce4208d213bb0760

macpijan committed:
u-boot: boot.cmd: adjust boot cmd

After update, uImage is built, not zImage.

Adjust U-Boot boot command accordingly to match this.

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
linux: enable obsolete GPIO_SYSFS

It's already obsolete interface, but we still rely on it, until we
switch to more modern gpiolib in:
https://github.com/3mdeb/RteCtrl/issues/21

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
rte-persistent-data: dummy recipe to create /data dir

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
wic/sunxi-mmc-spl.wks: bring back data partition

We need /data partition for persistent storage to preserve existing
features of RTE.

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
linux-mainline: fix spidev patch

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
bsp: flashrom: update to 1.4.0

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
connectivity: gensio: update to 2.8.7

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
connectivity: ser2net: update to 4.6.2

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
devtools: python3-asciinema: update to 2.4.0

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
tree-wide: fix license warnings

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
conf/distro/rte.conf: fix p4

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
rte-tests: add

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan commented:
@PLangowski @tym2k1 

I have fixed the problem with booting U-Boot (see: https://github.com/3mdeb/meta-rte/pull/73/commits/e0e6e4914623f338b3447237204c56ac7ae5f25b) and many more.

Added some tests (not present before).

Updated some crucial recipes (not done as part of the layer update).

Fixed some build warnings, etc.

This need full feature compatibility testing on real hardware. But I have tested it roughly, and the major remaining problem seems to be this one:

```
root@rte:~# dmesg | grep -i mcp
[    7.061151] mcp230xx 0-0020: error -ENXIO: can't write IOCON 32
```

The problem is this GPIO expander seems to not respond at all:

```
root@rte:~# i2cdetect -y 0
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:                         -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- --                         
root@rte:~# 
```

On another RTE (old version of OS) it can be probed:

```
root@rte:~# i2cdetect -y 0
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: UU -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- --    
```

and is present

```
cat /sys/class/gpio/gpiochip400/label 
mcp23017
```

Results from cukinia:

```
[PASS] Checking kernel config "GPIO_SYSFS" set to "y"
[PASS] Checking kernel config "SPI_SPIDEV" set to "y"
[PASS] Checking kernel config "PINCTRL_MCP23S08" set to "m"
[FAIL] Checking directory "/sys/class/gpio/gpiochip400" exists
[PASS] Checking directory "/sys/class/gpio/gpio199" exists
[FAIL] Checking directory "/sys/class/gpio/gpio11" exists
[FAIL] Checking directory "/sys/class/gpio/gpio12" exists
[PASS] Checking directory "/sys/class/gpio/gpio199" exists
[FAIL] Checking directory "/sys/class/gpio/gpio400" exists
[FAIL] Checking directory "/sys/class/gpio/gpio401" exists
[FAIL] Checking directory "/sys/class/gpio/gpio402" exists
[FAIL] Checking directory "/sys/class/gpio/gpio403" exists
[FAIL] Checking directory "/sys/class/gpio/gpio404" exists
[FAIL] Checking directory "/sys/class/gpio/gpio405" exists
[FAIL] Checking directory "/sys/class/gpio/gpio406" exists
[FAIL] Checking directory "/sys/class/gpio/gpio407" exists
[FAIL] Checking directory "/sys/class/gpio/gpio408" exists
[FAIL] Checking directory "/sys/class/gpio/gpio409" exists
[FAIL] Checking directory "/sys/class/gpio/gpio410" exists
[FAIL] Checking directory "/sys/class/gpio/gpio411" exists
[FAIL] Checking directory "/sys/class/gpio/gpio412" exists
[FAIL] Checking directory "/sys/class/gpio/gpio413" exists
[FAIL] Checking directory "/sys/class/gpio/gpio414" exists
[FAIL] Checking directory "/sys/class/gpio/gpio415" exists
[FAIL] Checking directory "/sys/class/gpio/gpio6" exists
[PASS] Checking "/dev/spidev1.0" exists and is character special
[PASS] Running "ser2net -v" is successful
[PASS] Running "flashrom --version" is successful
[PASS] Checking directory "/data" exists
[PASS] Checking "/dev/mmcblk0p1" exists and is block special
[PASS] Checking "/dev/mmcblk0p2" exists and is block special
[PASS] Checking "/dev/mmcblk0p3" exists and is block special
[PASS] Checking "/dev/mmcblk0p4" exists and is block special
[PASS] Running "findmnt /data" is successful
[PASS] Checking if systemd unit "ser2net.service" is active
[FAIL] Checking if systemd unit "RteCtrl.service" is active
result: 21 failure(s)
```



All failures are caused by the problem mentioned above.

macpijan commented:
UPDATE

The error from: https://github.com/3mdeb/meta-rte/pull/73#issuecomment-2433492993 happens on my HW also on old OS.  So likely my HW is damaged, and we should try this new image on another board.

m-iwanicki commented on conf/distro/rte.conf:
```diff
@@ -31,12 +31,10 @@ DISTRO_FEATURES_BACKFILL_CONSIDERED += "sysvinit pulseaudio"
 ### orange-pi-zero custom config ###
 ####################################
+PERSISTENT_DATA_DIR = "/data"
 IMAGE_FSTYPES_orange-pi-zero = "wic.gz wic.bmap"
```
```suggestion
IMAGE_FSTYPES:orange-pi-zero = "wic.gz wic.bmap"
```

Old override style

m-iwanicki commented:
@macpijan zImage error was because `KERNEL_IMAGETYPE` used old override style

```
KERNEL_IMAGETYPE_orange-pi-zero = "zImage"
```

@PLangowski fixed it but I think he forgot to push fix.
But I guess your fix also works

macpijan committed:
distro/rte.conf: fix remaining old style override syntax

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan commented on recipes-rte/rte-tests/files/cukinia.conf:
```diff
@@ -0,0 +1,47 @@
+cukinia_cmd ser2net -v
+cukinia_cmd flashrom --version
+cukinia_test -d /data
+# TODO: test for support for specific chips we care about (such as new ones for
```
@PLangowski Please fill this in, possibly also with other tests if you come up with something more.

macpijan replied:
@PLangowski This still needs to be done. We need to verify that flashrom supports all chips requested in the issue.

PLangowski commented:
@macpijan I'm getting this error (on a clean build):
```
Log data follows:
| DEBUG: Executing python function extend_recipe_sysroot
| NOTE: Direct dependencies are ['/build/../work/meta-sunxi/recipes-kernel/linux/linux-mainline_6.6.28.bb:do_populate_sysroot', '/build/../work/poky/meta/recipes-bsp/u-boot/u-boot_2024.01.bb:do_populate_sysroot', '/build/../work/poky/me
ta/recipes-core/glibc/cross-localedef-native_2.39.bb:do_populate_sysroot', '/build/../work/poky/meta/recipes-core/glibc/ldconfig-native_2.12.1.bb:do_populate_sysroot', '/build/../work/poky/meta/recipes-devtools/qemu/qemuwrapper-cross_1.
0.bb:do_populate_sysroot', '/build/../work/poky/meta/recipes-kernel/kmod/depmodwrapper-cross_1.0.bb:do_populate_sysroot', 'virtual:native:/build/../work/poky/meta/recipes-core/update-rc.d/update-rc.d_0.8.bb:do_populate_sysroot', 'virtua
l:native:/build/../work/poky/meta/recipes-devtools/createrepo-c/createrepo-c_1.0.4.bb:do_populate_sysroot', 'virtual:native:/build/../work/poky/meta/recipes-devtools/dnf/dnf_4.19.0.bb:do_populate_sysroot', 'virtual:native:/build/../work
/poky/meta/recipes-devtools/dosfstools/dosfstools_4.2.bb:do_populate_sysroot', 'virtual:native:/build/../work/poky/meta/recipes-devtools/fdisk/gptfdisk_1.0.9.bb:do_populate_sysroot', 'virtual:native:/build/../work/poky/meta/recipes-devt
ools/makedevs/makedevs_1.0.1.bb:do_populate_sysroot', 'virtual:native:/build/../work/poky/meta/recipes-devtools/mtools/mtools_4.0.43.bb:do_populate_sysroot', 'virtual:native:/build/../work/poky/meta/recipes-devtools/opkg-utils/opkg-util
s_0.6.3.bb:do_populate_sysroot', 'virtual:native:/build/../work/poky/meta/recipes-devtools/pseudo/pseudo_git.bb:do_populate_sysroot', 'virtual:native:/build/../work/poky/meta/recipes-devtools/rpm/rpm_4.19.1.1.bb:do_populate_sysroot', 'v
irtual:native:/build/../work/poky/meta/recipes-extended/parted/parted_3.6.bb:do_populate_sysroot', 'virtual:native:/build/../work/poky/meta/recipes-extended/pigz/pigz_2.8.bb:do_populate_sysroot', 'virtual:native:/build/../work/poky/meta
/recipes-support/bmaptool/bmaptool_git.bb:do_populate_sysroot']
| NOTE: Installed into sysroot: ['dosfstools-native', 'gptfdisk-native', 'mtools-native', 'parted-native']
| NOTE: Skipping as already exists in sysroot: ['linux-mainline', 'u-boot', 'base-files', 'base-passwd', 'gettext-minimal-native', 'cross-localedef-native', 'glibc', 'ldconfig-native', 'libxcrypt', 'systemd-systemctl-native', 'cmake-nat
ive', 'gcc-runtime', 'libgcc', 'libtool-native', 'm4-native', 'opkg-utils', 'qemu-native', 'qemuwrapper-cross', 'shadow-sysroot', 'shadow', 'texinfo-dummy-native', 'depmodwrapper-cross', 'linux-libc-headers', 'attr', 'ptest-runner', 'op
enssl-native', 'coreutils-native', 'expat-native', 'gettext-native', 'glib-2.0-native', 'libxml2-native', 'ncurses-native', 'readline-native', 'update-rc.d-native', 'util-linux-libuuid-native', 'util-linux-native', 'zlib-native', 'creat
erepo-c-native', 'desktop-file-utils-native', 'dnf-native', 'elfutils-native', 'file-native', 'flex-native', 'gnu-config-native', 'json-c-native', 'libcomps-native', 'libdnf-native', 'libedit-native', 'libmodulemd-native', 'librepo-nati
ve', 'lua-native', 'make-native', 'makedevs-native', 'opkg-utils-native', 'perl-native', 'pseudo-native', 'python3-build-native', 'python3-flit-core-native', 'python3-iniparse-native', 'python3-installer-native', 'python3-packaging-nati
ve', 'python3-pyproject-hooks-native', 'python3-setuptools-native', 'python3-six-native', 'python3-wheel-native', 'python3-native', 'rpm-native', 'swig-native', 'bzip2-native', 'libarchive-native', 'libidn2-native', 'libnsl2-native', 'l
ibsolv-native', 'libtirpc-native', 'lzlib-native', 'pigz-native', 'shadow-native', 'unzip-native', 'xz-native', 'zstd-native', 'gobject-introspection-native', 'kmod-native', 'acl-native', 'attr-native', 'bmaptool-native', 'curl-native',
 'debianutils-native', 'gdbm-native', 'gmp-native', 'gnutls-native', 'libtasn1-native', 'gpgme-native', 'itstool-native', 'libassuan-native', 'libbsd-native', 'libcap-ng-native', 'libcap-native', 'libcheck-native', 'libffi-native', 'lib
gcrypt-native', 'libgpg-error-native', 'libmd-native', 'libmicrohttpd-native', 'libpcre2-native', 'libunistring-native', 'libyaml-native', 'nettle-native', 'popt-native', 'shared-mime-info-native', 'sqlite3-native']
| DEBUG: Python function extend_recipe_sysroot finished
| DEBUG: Executing python function set_image_size
| DEBUG: 291028.400000 = 223868 * 1.300000
| DEBUG: 397524.400000 = max(291028.400000, 8192)[291028.400000] + 106496
| DEBUG: 397525.000000 = int(397524.400000)
| DEBUG: 397525 = aligned(397525)
| DEBUG: returning 397525
| DEBUG: Python function set_image_size finished
| DEBUG: Executing shell function do_image_wic
| /work/poky/scripts/lib/wic/engine.py:362: SyntaxWarning: invalid escape sequence '\/'
|   abs_path = re.sub('\/\/+', '/', path)
| INFO: Creating image(s)...
|
| WARNING: bootloader config not specified, using defaults
|
| INFO: Intentionally disabled removing partition images.
| ERROR: A native program mkfs.ext4 required to build the image was not found (see details above).
|
| Please make sure wic-tools have e2fsprogs-native in its DEPENDS, build it with 'bitbake wic-tools' and try again.
|
|
| WARNING: exit code 1 from a shell command.
ERROR: Task (/build/../work/poky/meta/recipes-core/images/core-image-minimal.bb:do_image_wic) failed with exit code '1'
NOTE: Tasks Summary: Attempted 5283 tasks of which 13 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /build/../work/poky/meta/recipes-core/images/core-image-minimal.bb:do_image_wic
Summary: There were 32 WARNING messages.
Summary: There was 1 ERROR message, returning a non-zero exit code.
2024-10-25 07:25:02 - ERROR    - Command "/work/poky/bitbake/bin/bitbake -c build core-image-minimal core-image-minimal-swu" failed with error 1
```

EDIT: 
Fixed by adding `e2fsprogs-native` to the `WKS_FILE_DEPENDS`

PLangowski committed:
conf/distro/rte.conf: Add e2fsprogs-native to WKS_FILE_DEPENDS

Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>

PLangowski committed:
recipes-core/images/core-image-minimal: Uncomment root password settings

Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>

PLangowski commented:
Something very unusual is happening.

Booting hangs on  `[    1.769553] Waiting for root device /dev/mmcblk0p2...` and never continues. It doesn't fall back to shell either.

First we thought that the error can be resolved by specifying the partition UUID in the boot command:

```
setenv bootargs console=${console} console=tty1 root=PARTUUID=076c4a2a-02 rootwait panic=10 ${extra}
```

Thisfixes the problem with mounting the root partition, but booting still does not complete. It hangs on:
```
[    7.091486] dwmac-sun8i 1c30000.ethernet end0: configuring for phy/mii link mode
[    7.184108] random: crng init done
```
And falls back to recovery shell.
We tried increasing log verbosity to debug the issue. Here's where the weird stuff happened: The system booted. We then tried booting it with unchanged `root` parameter. It also booted. For some reason adding debug-related options fixes the issue. We have no idea why. Below are the steps required to boot the system:

1. Power on the RTE and go interrupt booting to get into U-boot shell
2.  In U-boot shell execute the following commands:
```
setenv boot_part 2
setenv bootargs console=${console} console=tty1 root=/dev/mmcblk0p${boot_part} rootwait panic=10 ${extra} ignore_loglevel initcall_debug
ext4load mmc 0:${boot_part} ${fdt_addr_r} /boot/${fdtfile}
ext4load mmc 0:${boot_part} ${kernel_addr_r} /boot/uImage
booti ${kernel_addr_r} - ${fdt_addr_r} || bootm ${kernel_addr_r} - ${fdt_addr_r}
```

The commands are taken from `recipes-bsp/u-boot/u-boot/orange-pi-zero/boot.cmd`. The only difference is the presence of `ignore_loglevel initcall_debug`.

macpijan commented:
```
[    7.091486] dwmac-sun8i 1c30000.ethernet end0: configuring for phy/mii link mode
[    7.184108] random: crng init done
```

This is probably unrelated log. Look in `dmesg` or `journactl -x` for erros.

Check `mount --all` in recovery shell and check if it does not complain.

Check `fstab`. Do you have all partitions `/dev/mmcblk0p[1-4]` visible in OS?

I can try removing my `tmp`, rebuild and test again on my unit.

macpijan commented:
In the presence of these "weirdnesses you may want to check on different hardware unit as well.

macpijan commented:
@PLangowski I was able to trigger the `Waiting for root device /dev/mmcblk0p2...` problem as well. Your conclusions are incorrect. This problem is completely random. Just try booting the board 10 times for instance with no changes to the boot commands.

Sometimes the SD card gets enumerated as mmcblk0, sometimes as mmcblk1. There is a race condition, the SDIO WiFi chip sometimes gets enumerated first as mmc0, and the SD card gets mmc1 then. We do not use WiFi, so we could even remove it (disable in devicetree) as a quick workaround to this issue.

https://paste.dasharo.com/?a383dc8cd91aca31#2Mt88gsFh2o1xQcnMw2HMVoPGLBv4122eX5JFxD5hEvQ

Let's focus on SPI first here.

PLangowski commented:
@macpijan The GPIO expander pins have been shifted by 112. I don't know the reason why this happened. Some service tries to export `GPIO 404` during late boot, but fails to do so. We should probably modify that so that it exports 516, 517 and 518.

PLangowski committed:
recipes-kernel: linux-mainline: Add CONFIG_DEBUG_FS

Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>

m-iwanicki commented on recipes-core/images/core-image-minimal.bbappend:
```diff
@@ -1,16 +1,14 @@
+    locale-base-en-us \
+    "
 
+INHERIT += "extrausers"
```
@PLangowski 

> [INHERIT](https://docs.yoctoproject.org/ref-manual/variables.html#term-INHERIT)[](https://docs.yoctoproject.org/ref-manual/variables.html#term-INHERIT)
>
>    Causes the named class or classes to be inherited globally. Anonymous functions in the class or classes are not executed for the base configuration and in each individual recipe. **The OpenEmbedded build system ignores changes to [INHERIT](https://docs.yoctoproject.org/ref-manual/variables.html#term-INHERIT) in individual recipes**

I quoted most important part in **bold**. You should probably move this to distro conf file or use

```
inherit extrausers
```

PLangowski replied:
26663e11e20251cd683c4b0770d8e9c00541114a

PLangowski committed:
recipes-core/images/core-image-minimal.bbappend: Use encrypted root password

Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>

PLangowski committed:
recipes-go: RteCtrl.cfg: Modify GPIO expander numbers

The GPIO expander pin numbers have recently been shifted by 112. Therefore, the
numbers in RteCtrl needed to be adjusted.

Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>

PLangowski commented:
Fixed the SPI problem by modifying the RteCtrl config.

Now after boot the GPIOs are correctly exported and can be used to control SPI.

PLangowski committed:
recipes-kernel/linux/linux-mainline: Disable WiFi chip

Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>

PLangowski commented:
I also disabled the WiFi chip. It seems that we fixed all the problems.

macpijan commented on recipes-kernel/linux/linux-mainline/0010-sun8i-h2-plus-orangepi-zero.dts-disable-mmc1.patch:
```diff
@@ -0,0 +1,26 @@
+Date: Tue, 29 Oct 2024 09:47:51 +0100
+Subject: [PATCH] Disable wifi chip
+
+Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>
```
Upstream-Status could be added

m-iwanicki replied:
We probably need to add status to all patches. I wonder why build didn't error out as it usually does without Upstream-Status

PLangowski replied:
[b089531](https://github.com/3mdeb/meta-rte/pull/73/commits/b089531c326b298eb1392b134a70cf4768ad12dc)

macpijan replied:
It's not really oe-specific, rather RTE specific, but ok

macpijan commented on recipes-kernel/linux/linux-mainline/0010-sun8i-h2-plus-orangepi-zero.dts-disable-mmc1.patch:
```diff
@@ -0,0 +1,26 @@
+From: Pawel Langowski <pawel.langowski@3mdeb.com>
+Date: Tue, 29 Oct 2024 09:47:51 +0100
+Subject: [PATCH] Disable wifi chip
+
```
It would be great if we develop a habit of adding some more context on **why** certain changes are introduced.

In this particular case, it could be added in the patch itself, or in the commit text. We've even had some relevant discussion in GH MR that could be linked there as a reference to this explanation. It does not have to be long, just a sentence or two can tremendously improve understanding of commits if we go back to them after a longer time period.

PLangowski replied:
[b089531](https://github.com/3mdeb/meta-rte/pull/73/commits/b089531c326b298eb1392b134a70cf4768ad12dc)

macpijan replied:
Thanks

macpijan reviewed (changes requested):
This commit https://github.com/3mdeb/meta-rte/pull/73/commits/e3937c931cd59ce6605b89601d800f58e86a60d1 could use some explanation as well.

Also, it's name is rather long, we do not always need to include the full path.

macpijan commented on recipes-go/3mdeb-rtectrl/3mdeb-rtectrl/RteCtrl.cfg:
```diff
@@ -21,112 +21,112 @@
             "direction" : "out",
             "init_value" : 0,
-            "sys_gpio" : 404
+            "sys_gpio" : 516
```
Do we know why this has changed?

PLangowski replied:
Unfortunately I don't know

macpijan commented on recipes-kernel/linux/linux-mainline/debug-fs.cfg:
```diff
@@ -0,0 +1,26 @@
+# CONFIG_GENERIC_IRQ_DEBUGFS is not set
```
Are you sure we need those `is not set` entries?
If we omit them, are these set to `y`?

m-iwanicki replied:
Those configs are likely generated automatically do we want to then modify them by hand?
It takes time to check whether these options can be safely removed and increases chances of mistakes

macpijan replied:
My question was a bit rhetorical.

What `diffconfig` does, it generates a diff in `.config` files **before** and **after** you have enabled some options in the `menuconfig`.

If new `is not set` entries appear in the **second** config when you enabled some option, it simply means that new options became available (they simply have not been present at all in the **first** config due to unmet dependencies) and they **are** disabled by default. There is no room for mistake here - unless you only enabled some options in one `diffconfig` iteration, not enable some, and disable some.

I agree it's not huge, but it's simply not needed to leave these `is not set` entries. It may also trigger some conflicts as well later on, if we decide to enable some features in upcoming fragments. Lastly, it likely would not be accepted upstream.

TLDR: unless you have only enabled some options and used `diffconfig`, it is safe (and recommended) to remove all `is not set` options. If you specifically disabled some option (which is rather uncommon thing to do, in fragments we mostly enable some features), it should stay as disabled.

PLangowski replied:
Done [c566016](https://github.com/3mdeb/meta-rte/pull/73/commits/c5660163e0dd179f93452a4884bcf04490a05f6a)

macpijan commented on recipes-rte/rte-tests/files/cukinia.conf:
```diff
@@ -0,0 +1,47 @@
+cukinia_test -d /sys/class/gpio/gpio11
+cukinia_test -d /sys/class/gpio/gpio12
+cukinia_test -d /sys/class/gpio/gpio199
+cukinia_test -d /sys/class/gpio/gpio400
```
@PLangowski These must be updated

PLangowski replied:
bc6884190301773097e97b3ddb40e133d1f39f30

PLangowski committed:
recipes-rte: cukinia.conf: Change GPIO numbers

Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>

PLangowski commented:
@macpijan I added the explanation [93fb19f](https://github.com/3mdeb/meta-rte/pull/73/commits/93fb19fb884e708ba1bda61297c8e11e7e2de2b3)

macpijan commented:
Looks good enough for me to get merged into develop.

This thread bother me though: https://github.com/3mdeb/meta-rte/pull/73#discussion_r1824153506

In the end, we must understand this.

pietrushnic commented:
> Some service tries to export GPIO 404 during late boot, but fails to do so. 

@PLangowski @macpijan, which service does it? I have a problem with the homelab setup of RTE, where SPI is not enabled at boot on v0.7.5. I have no idea why it happens, and you seem to have a related discussion in this MR.

macpijan commented:
> @PLangowski @macpijan, which service does it?

It's `RteCtrl.service` IIRC

pietrushnic commented:
@macpijan as I'm trying to describe in [this Dasharo OSFV thread](https://matrix.to/#/!MwWOJhMJzlxIdAQxae:matrix.3mdeb.com/$QxB0WW3IWX4Ffyb_Q-cG81bm3taf0MwkQbKFo5P4ur0?via=matrix.3mdeb.com&via=matrix.org&via=nitro.chat). For now I just changed `/etc/RteCtrl/RteCtrl.cfg`
